api/middleware/stats.go

```diff
@@ -0,0 +1,83 @@
+package middleware
+
+import (
+	"fmt"
+	"github.com/raintank/metrictank/stats"
+	"path"
+	"strings"
+	"sync"
+)
+
+type requestStats struct {
+	sync.Mutex
+	responseCounts    map[string]map[int]*stats.Counter32
+	latencyHistograms map[string]*stats.LatencyHistogram15s32
+	sizeMeters        map[string]*stats.Meter32
+}
+
+func (r *requestStats) PathStatusCount(path string, status int) {
+	r.Lock()
+	p, ok := r.responseCounts[path]
+	if !ok {
+		p = make(map[int]*stats.Counter32)
+		r.responseCounts[path] = p
+	}
+	c, ok := p[status]
+	if !ok {
+		c = stats.NewCounter32(fmt.Sprintf("api.request.%s.%d", pathSlug(path), status))
+		p[status] = c
+	}
+	r.Unlock()
+	c.Inc(1)
+}
+
+func (r *requestStats) PathLatency(path string, dur time.Duration) {
+	r.Lock()
+	p, ok := r.latencyHistograms[path]
+	if !ok {
+		p = stats.NewLatencyHistogram12h32(fmt.Sprintf("api.request.%s.latency", pathSlug(path)))
```

Comment:
latencyhistograms already add a prefix like `.latency.min.gauge32` to the metric name.